internal/devices/create.go

```diff
@@ -64,62 +66,127 @@ func (c *Client) Create() *cobra.Command {
   metal device create -p $METAL_PROJECT_ID -P c3.medium.x86 -m sv -H test-rocky -O rocky_8 -r 47161704-1715-4b45-8549-fb3f4b2c32c7`,
 
 		RunE: func(cmd *cobra.Command, args []string) error {
-			var endDt *packngo.Timestamp
-
 			if userdata != "" && userdataFile != "" {
 				return fmt.Errorf("either userdata or userdata-file should be set")
 			}
 			cmd.SilenceUsage = true
-
 			if userdataFile != "" {
 				userdataRaw, readErr := os.ReadFile(userdataFile)
 				if readErr != nil {
-					return fmt.Errorf("Could not read userdata-file: %w", readErr)
+					return fmt.Errorf("could not read userdata-file: %w", readErr)
 				}
 				userdata = string(userdataRaw)
 			}
+			var endDt time.Time
 
 			if terminationTime != "" {
 				parsedTime, err := time.Parse(time.RFC3339, terminationTime)
 				if err != nil {
-					return fmt.Errorf("Could not parse time %q: %w", terminationTime, err)
+					return fmt.Errorf("could not parse time %q: %w", terminationTime, err)
 				}
-				endDt = &packngo.Timestamp{Time: parsedTime}
+				endDt = parsedTime
+			}
+			var publicSubnet int32
+			// In case of no configuration,
+			if publicIPv4SubnetSize == 0 {
+				publicSubnet = 31 // Assigin default to 31
+			} else {
+				publicSubnet = int32(publicIPv4SubnetSize)
 			}
 
+			formatedCdata := make(map[string]interface{})
+			if customdata != "" {
+				err := json.Unmarshal([]byte(customdata), &formatedCdata)
+				if err != nil {
+					return fmt.Errorf("could not convert customedata :%w", err)
+				}
+			}
 			var facilityArgs []string
+
+			var request metal.ApiCreateDeviceRequest
 			if facility != "" {
 				facilityArgs = append(facilityArgs, facility)
-			}
 
-			request := &packngo.DeviceCreateRequest{
-				Hostname:              hostname,
-				Plan:                  plan,
-				Facility:              facilityArgs,
-				Metro:                 metro,
-				OS:                    operatingSystem,
-				BillingCycle:          billingCycle,
-				ProjectID:             projectID,
-				UserData:              userdata,
-				CustomData:            customdata,
-				IPXEScriptURL:         ipxescripturl,
-				Tags:                  tags,
-				PublicIPv4SubnetSize:  publicIPv4SubnetSize,
-				AlwaysPXE:             alwaysPXE,
-				HardwareReservationID: hardwareReservationID,
-				SpotInstance:          spotInstance,
-				SpotPriceMax:          spotPriceMax,
-				TerminationTime:       endDt,
+				facilityDeviceRequest := metal.CreateDeviceRequest{
+					DeviceCreateInFacilityInput: &metal.DeviceCreateInFacilityInput{
+						Facility:        facilityArgs,
+						Plan:            plan,
+						OperatingSystem: operatingSystem,
+						Hostname:        &hostname,
+					},
+				}
+
+				if billingCycle != "" {
+					facilityDeviceRequest.DeviceCreateInFacilityInput.SetBillingCycle(billingCycle)
+				}
+				if userdata != "" && userdataFile != "" {
+					facilityDeviceRequest.DeviceCreateInFacilityInput.SetUserdata(userdata)
```

Comment:
Do we need to break out this `SetUserdata` call instead of setting user data when the `facilityDeviceRequest` is declared?  Does something bad happen if we send `"userdata": ""` to the API?

Reply:
It working fine. will remove it. 

```
                                                Hostname:        &hostname,
+                                               Userdata:        &userdata,
                                        },
                                }

                                if billingCycle != "" {
                                        facilityDeviceRequest.DeviceCreateInFacilityInput.SetBillingCycle(billingCycle)
                                }
-                               if userdata != "" && userdataFile != "" {
-                                       facilityDeviceRequest.DeviceCreateInFacilityInput.SetUserdata(userdata)
-                               }
+                               // if userdata != "" && userdataFile != "" {
+                               //      facilityDeviceRequest.DeviceCreateInFacilityInput.SetUserdata(userdata)
+                               // }
                                if alwaysPXE {
                                        facilityDeviceRequest.DeviceCreateInFacilityInput.SetAlwaysPxe(alwaysPXE)
                                }
@@ -152,14 +153,15 @@ func (c *Client) Create() *cobra.Command {
                                                Plan:            plan,
                                                OperatingSystem: operatingSystem,
                                                Hostname:        &hostname,
+                                               Userdata:        &userdata,
                                        },
                                }
                                if billingCycle != "" {
                                        metroDeviceRequest.DeviceCreateInMetroInput.SetBillingCycle(billingCycle)
                                }
-                               if userdata != "" && userdataFile != "" {
-                                       metroDeviceRequest.DeviceCreateInMetroInput.SetUserdata(userdata)
-                               }
+                               // if userdata != "" && userdataFile != "" {
+                               //      metroDeviceRequest.DeviceCreateInMetroInput.SetUserdata(userdata)
+                               // }
```